plugin/pkg/admission/runtimeclass/admission.go

```diff
@@ -91,8 +91,12 @@ func (r *RuntimeClass) Admit(attributes admission.Attributes, o admission.Object
 		return err
 	}
 	if utilfeature.DefaultFeatureGate.Enabled(features.PodOverhead) {
-		err = setOverhead(attributes, pod, runtimeClass)
-		if err != nil {
+		if err := setOverhead(attributes, pod, runtimeClass); err != nil {
+			return err
+		}
+	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.RuntimeClass) {
```

Comment:
Move this check to the top of the function (short circuit). The controller shouldn't run if RuntimeClass isn't enabled.

Reply:
done, please take another look